tangelo/linq/translator/translate_qubitop.py

```diff
@@ -76,7 +76,7 @@ def translate_operator(qubit_operator, source, target, n_qubits=None):
         if target in {"qiskit", "sympy"}:
             # The count_qubits function has no way to detect the number of
             # qubits when an operator is only a tensor product of I.
-            if qubit_operator == QubitOperator((), qubit_operator.constant):
+            if qubit_operator == QubitOperator((), qubit_operator.constant) and n_qubits is None:
```

Comment:
This does not flow well with L81, that says " if n_qubits is None, then its value is count_qubits(q_op)".

Either that conditional statement at L81 is not necessary anymore (since it will never be encountered as L79 prevents it) or L79 is not the solution.

We need the case for both sympy and qiskit to work, make sure the behaviour is as expected for them (with or without n_qubits).

Reply:
This code raises `ValueError: The number of qubits (n_qubits) must be provided.`, even if I provided `n_qubits`. If we don't check for `n_qubits`, it will fail for every empty `QubitOperator`. The `count_qubits` can automatically get the number of qubits of an operator, but it fails with an empty `QubitOperator`.

```python
from tangelo.linq import translate_operator
from tangelo.toolboxes.operators import QubitOperator

op = QubitOperator((), 5.)
translate_operator(op, source="tangelo", target="qiskit", n_qubits=5)
```

Reply:
Thank you. If that means all use cases are covered and handled as they should, then you can resolve this conversation :)